### PR TITLE
Using SSH_AUTH_SOCK (ssh agent forwarding) to pull upm private repos

### PR DESCRIPTION
**Pull request by ivan-hernandez-scopely**
#### Changes

- This PR is a proposal to fix this issue: https://github.com/game-ci/unity-builder/issues/161
- In order to get the SSH_AUTH_SOCK we used https://github.com/webfactory/ssh-agent/issues/11
- sshAgent is an optional input parameter.

## Usage
```

      - name: Setup SSH Agent
        uses: webfactory/ssh-agent@v0.5.2
        with:
          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}

      - name: Unity Builder
        uses: ivan-hernandez-scopely/unity-builder@feature/ssh-agent-for-private-upm
        env:
          UNITY_EMAIL: ${{ secrets.UNITY_EMAIL }}
          UNITY_PASSWORD: ${{ secrets.UNITY_PASSWORD }}
          UNITY_SERIAL: ${{ secrets.UNITY_SERIAL }}
        with:
          projectPath: ${{ matrix.projectPath }}
          targetPlatform: ${{ matrix.targetPlatform }}
          sshAgent: ${{ env.SSH_AUTH_SOCK }}
```

#### Checklist

- [x] Read the contribution [guide](../CONTRIBUTING.md) and accept the [code](../CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

**Comment by webbertakken** [comment]
This looks like a great approach to me.

The hardest part about merging this is actually figuring out whether this would be the default approach that we'll support from here on forward (or risk bc break in future versions) @davidmfinol @GabLeRoux @frostebite please let me know your thoughts on this.

**Comment by webbertakken** [comment]
What made you chose this approach over mounting `~/.ssh/` (which is an existing mount)?

File: dist/Dockerfile [hunk 1]
```diff
@@ -17,4 +17,6 @@ ADD entrypoint.sh /entrypoint.sh
 RUN chmod +x /entrypoint.sh
 RUN ls
 
+RUN apt-get update && apt-get install -y openssh-client
```
[the remark this answers is not in the document]
Comment (frostebite) [reply dist/Dockerfile:20]: @webbertakken is there any alternative? My understanding is this demonstrates a situation we simply need it?

[the remark this answers is not in the document]
Comment (webbertakken) [reply dist/Dockerfile:20]: I agree. I think we need this, but not as part of builder. This is why I'm suggesting to move it to the base image.

The base image installs everything with the flag `--no-install-recommends`, to keep the image as small as possible.

File: dist/Dockerfile [hunk 2]
```diff
@@ -17,4 +17,6 @@ ADD entrypoint.sh /entrypoint.sh
...
+
```
Comment (webbertakken) [inline dist/Dockerfile:21]: This change needs to be done upstream. See https://github.com/game-ci/docker/pull/117
```suggestion
```

**Comment by webbertakken** [comment]
Note that this will start working only after 0.14 of the docker images has been released.

**Comment by uchar** [comment]
@ivan-hernandez-scopely Does your solution works with private GitHub repositories?

**Comment by ivan-hernandez-scopely** [comment]
> @ivan-hernandez-scopely Does your solution works with private GitHub repositories?

@uchar yes. Notice that you need to provide the ssh private key to the webfactory/ssh-agent as a github secret:
  - https://github.com/XXX/YYY/settings/secrets/actions

And setup its ssh public key in:
  - https://github.com/settings/keys

**Comment by uchar** [comment]
@webbertakken  When does the new version going to release?

**Comment by webbertakken** [comment]
@uchar I should have time in the weekend.

In the meantime you can use the specific commit hash until the new version is there.

**Comment by ivan-hernandez-scopely** [comment]
> @uchar I should have time in the weekend.
> 
> In the meantime you can use the specific commit hash until the new version is there.

But this commit hash will require the docker images 0.14 released. right?

**Comment by webbertakken** [comment]
Oh yes, you're right. I was waiting for https://github.com/game-ci/docker/pull/116 to be merged for that. It's currently pending a last change.

**Comment by uchar** [comment]
@webbertakken Can you please release the new version, I really need this :pray:

**Comment by webbertakken** [comment]
Thanks for the heads-up. Indeed, the other PR doesn't seem to be moving as fast as we'd hoped.

I just released a new version for both docker and builder. Please follow the progress of the docker images on our [image versions page](https://game.ci/docs/docker/versions). Please allow up to 24 hours for all of them to be published.

**Pull request merged**